src/Build/Definition/ProjectCollection.cs

```diff
@@ -341,6 +341,9 @@ public ProjectCollection(IDictionary<string, string> globalProperties, IEnumerab
                 {
                     _globalProperties = new PropertyDictionary<ProjectPropertyInstance>(globalProperties.Count);
 
+                    // at this stage globalProperties collection contains entries passed from command line (e.g. /p:foo=bar).
```

Comment:
I'm a slow thinker - so it's not clear to me, why there cannot be any global properties set either from API or from call to MSBuild task with explicit specification of global props (ot not sure if there is yet another way to set global properties) - can you make sure it's really just from commandline and possibly add the reasoning in the comment?